[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82584
* __->__ #81322
* #81321
* #81320
* #81319

This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

Commit:
[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)

This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

aaronenyeshi reviewed (approved):
LGTM! Also reviewed internally.

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

robieta commented:
@pytorchbot merge -l

Commit:
Update on "[Profiler] Make KinetoEvent a view of Result (Part 4 (final), stragglers)"


This PR just moves all the KinetoEvent methods which didn't fit the previous categories. Now that we no longer need to set Kineto event fields in `EventFieldsVisitor` we can remove the reference wrapper and rename the visitor to `AddKinetoMetadata` since that's all it does now.

Differential Revision: [D37490053](https://our.internmc.facebook.com/intern/diff/D37490053/)

[ghstack-poisoned]

robieta commented:
@pytorchbot merge -l

robieta commented:
@pytorchbot merge -f "I have tried to land this responsibly but infra is too flaky."
